packages/edge/src/middleware-helpers.ts

```diff
@@ -1,9 +1,39 @@
+export interface ModifiedRequest {
+  /**
+   * If this is set, the request headers will be overridden with this value.
+   */
```

Comment:
Let's add the examples in the docs that show that the original headers are most likely to be cloned and passed modified (if I got this correctly):

```suggestion
  /**
   * If set, will overwrite the incoming headers to the origin request.
   *
   * This is useful when you want to pass data between a Middleware and a
   * Serverless or Edge Function.
   *
   * @example
   * <caption>Add a `x-user-id` header and remove the `Authorization` header</caption>
   *
   * ```ts
   * import { rewrite } from '@vercel/edge';
   * export default async function middleware(request: Request): Promise<Response> {
   *   const newHeaders = new Headers(request.headers);
   *   newHeaders.set('x-user-id', 'user_123');
   *   newHeaders.delete('authorization');
   *   return rewrite(request.url, {
   *     request: { headers: newHeaders }
   *   })
   * }
   * ```
   */
```